src/Avalonia.Controls.DataGrid/Utils/DataGridHelper.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Avalonia.Controls
+{
+    public static class DataGridHelper
+    {
+        internal static void SyncColumnProperty<T>(AvaloniaObject column, AvaloniaObject content, AvaloniaProperty<T> property)
+        {
+            SyncColumnProperty(column, content, property, property);
+        }
+
+        internal static void SyncColumnProperty<T>(AvaloniaObject column, AvaloniaObject content, AvaloniaProperty<T> contentProperty, AvaloniaProperty<T> columnProperty)
```

Comment:
It may looks not really good. It also could be replaced with bindings between cell content and column.
But we probably want to keep implementation closer to the WPF and WindowsCommunityToolkit, so it should be easier to maintain.
For reference: https://github.com/dotnet/wpf/blob/master/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridHelper.cs#L675

Reply:
Probably worth marking the `DataGridHelper` class as `internal` btw?

Reply:
Makes sense, thanks